Feature/86 refactor text field component

thomastepi opened the pull request:
Refactored TextFieldComponents to enhance customization by introducing the CustomTextField component. This change allows each rendered text field to be modified with specific props. Additionally, PropTypes have been added to CustomTextField to enforce prop validation.

**Changes Made:**

1.  Implemented CustomTextField component to replace direct usage of MUI TextField.
2.  Introduced PropTypes to validate props passed to CustomTextField.
3.  Created test cases to verify component functionality.

**Reason for Changes:**

To streamline component reusability and maintain consistency in prop usage throughout the application.

erenfn reviewed (changes requested):
- The default font size should be 13px.
- Also, please adjust the height of the text field accordingly.
- It would be very helpful if you could provide a brief explanation of the prop types as a block comment in the file.
- Some details, such as shadows, were missing. You can find these in the Style Guide Figma file.
- It would be good if you could specify the default parameters.

Other than that, good work!

erenfn commented on frontend/src/tests/scenes/login/LoginPage.test.jsx:
```diff
@@ -1,41 +1,40 @@
-import { describe, it, expect } from 'vitest';
-import { render, screen, fireEvent } from '@testing-library/react';
-import LoginPage from '../../../scenes/login/LoginPage';
+import { describe, it, expect } from "vitest";
```
Also why do you have these files? I think they are not merged into the develop branch

thomastepi commented:
> * The default font size should be 13px.
> * Also, please adjust the height of the text field accordingly.
> * It would be very helpful if you could provide a brief explanation of the prop types as a block comment in the file.
> * Some details, such as shadows, were missing. You can find these in the Style Guide Figma file.
> * It would be good if you could specify the default parameters.
> 
> Other than that, good work!

I have implemented the changes you recommended and also updated the file to align with the styling in the Figma guide. thanks

swoopertr commented on frontend/src/components/TextFieldComponents/CustomTextField.jsx:
```diff
@@ -0,0 +1,133 @@
+  );
+};
+
+// Explanation of the PropTypes used in this file
```
those descriptions should be in readme.md file with an appropriate explanation.

swoopertr commented on frontend/src/components/TextFieldComponents/CustomTextField.jsx:
```diff
@@ -0,0 +1,133 @@
+      <InputLabel className="label" sx={{ fontWeight: labelFontWeight }}>
+        {labelText}
+      </InputLabel>
+      <TextField
```
this component usage is too complex, hard to understand, and difficult to modify. Please split the more basic functions or objects.

swoopertr commented on frontend/src/components/TextFieldComponents/TextFieldComponents.jsx:
```diff
@@ -1,93 +1,111 @@
           defaultValue="olivia@untitledui.com"
           error
           helperText="This is an error message."
+          endAdornment=<ErrorOutlineIcon color="error" />
```
What are the colors ? "error" error color?
can you make the selection with enum like; ErrorOutlineIcon.Color.Error

thomastepi commented on frontend/src/components/TextFieldComponents/CustomTextField/CustomTextField.jsx:
```diff
@@ -0,0 +1,96 @@
+  inputHeight: "56px",
+};
+
+CustomTextField.propTypes = {
```
should i delete the prop types

erenfn commented on frontend/src/components/TextFieldComponents/CustomTextField/CustomTextField.jsx:
```diff
@@ -0,0 +1,96 @@
+  );
+};
+
+CustomTextField.defaultProps = {
```
Vlad means this part

erenfn commented on frontend/src/components/TextFieldComponents/CustomTextField/CustomTextField.jsx:
```diff
@@ -0,0 +1,86 @@
+import "./CustomTextFieldStyles.css";
+
+const CustomTextField = ({
+  labelText,
```
Add these defaults here

  labelText: "",
  defaultValue: "",
  helperText: "",
  error: false,
  multiline: false,
  rows: 1,
  chips: [],
  startAdornment: null,
  endAdornment: null,
  placeholder: "",
  labelFontWeight: 600,
  TextFieldWidth: "100%",
  inputHeight: "56px",
